File: scaffolder-templates/dart_template/microservice/lib/modules/users/service/user_service.dart

import 'package:microservice/modules/users/entity/user.dart';
import 'package:microservice/modules/users/repository/user_repository.dart';
import 'package:shelf_modular/shelf_modular.dart';
import 'package:simple_logger/simple_logger.dart';

class UserService {
  UserRepository repository;

  SimpleLogger logger;

  UserService(this.repository, this.logger);

  Future<List<User>> getUsers() {
    logger.info('searching for users');
    return repository.getAllUsers();
  }
}

File: scaffolder-templates/dotnet-core-template/microservice/microservice/Service/UserService.cs
namespace microservice.Service;

using microservice.Data;
using microservice.Entity;

public class UserService {
    private MicroserviceContext context;

    private ILogger<UserService> logger;

    public UserService(MicroserviceContext context, ILogger<UserService> logger) {
        this.context = context;
        this.logger = logger;
    }

    public List<User> getUsers() {
        logger.LogInformation("retrieving users");
        return context.Users.ToList();
    }
}